src/vs/workbench/electron-main/main.ts

```diff
@@ -253,7 +255,11 @@ function setupIPC(): TPromise<Server> {
 // and assign them to the process environment (e.g. when doubleclick app on Mac)
 getUserEnvironment()
 	.then(userEnv => {
-		assign(process.env, userEnv);
+		if (!tty.isatty(1)) {
+			// Only assign global environment variables when
+			// STDOUT is not attached to terminal.
+			assign(process.env, userEnv);
```

Comment:
Instead of testing if vscode is run from terminal, different `assign` method could be used. Maybe:

```js
assign(userEnv, process.env);
process.env = userEnv;
```